Packages/MIES/MIES_ForeignFunctionInterface.ipf

```diff
@@ -142,3 +142,65 @@ Function/WAVE FFI_QueryLogbookUniqueSetting(string device, variable logbookType,
 
 	return settings
 End
+
+/// @brief Save the given permanent Igor Pro `datafolder` in the HDF5 format into `filepath`
+Function FFI_SaveDataFolderToHDF5(string filepath, string datafolder)
+
+	return FFI_SaveDataFolderToHDF5Impl(filepath, {datafolder})
+End
+
+static Function FFI_SaveDataFolderToHDF5Impl(string filepath, WAVE/T datafolders)
+
+	variable ref = NaN
+	variable groupID
+	string entry, cleanName
+
+	ASSERT(!FileExists(filepath), "filepath points to an existing file")
+	ASSERT(!FolderExists(filepath), "filepath points to an existing folder")
```

Comment:
@t-b, should the error message read: filepath needs a file name? In other words, it's not possible to reach this assertion unless the file name is absent from the path?

Reply:
Thanks. Rephrased.